pkg/api/types.go

```diff
@@ -2263,3 +2263,83 @@ type ThirdPartyResourceData struct {
 
 	Data []byte `json:"name,omitempty"`
 }
+
+// PodSecurityPolicy governs the ability to make requests that affect the SecurityContext
+// that will be applied to a container.
+type PodSecurityPolicy struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Privileged determines if a container can request to be run as privileged.
+	Privileged bool `json:"privileged,omitempty"`
+	// Capabilities is a list of capabilities that can be requested to add to the container.
+	Capabilities []Capability `json:"capabilities,omitempty"`
+	// HostPath determines if the policy allows containers to use the HostDir volume plugin
+	HostPath bool `json:"hostPath,omitempty"`
+	// HostNetwork determines if the policy allows the use of HostNetwork in the pod spec.
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// HostPorts determines if the policy allows host ports in the containers.
+	HostPorts bool `json:"hostPorts,omitempty"`
+	// SELinuxContext is the strategy that will dictate what labels will be set in the SecurityContext.
+	SELinuxContext SELinuxContextStrategyOptions `json:"seLinuxContext,omitempty"`
```

Comment:
Yes you should be able to get less than your policy allows (as a user, you
can choose not to delete things your policy allows you to delete).

On Sep 14, 2015, at 5:53 PM, Mike Danese <notifications@github.com> wrote:

In pkg/api/types.go
<https://github.com/kubernetes/kubernetes/pull/7893#discussion_r39452293>:

> +type PodSecurityPolicy struct {
> +	TypeMeta   `json:",inline"`
> +	ObjectMeta `json:"metadata,omitempty"`
> +
> +	// Privileged determines if a container can request to be run as privileged.
> +	Privileged bool `json:"privileged,omitempty"`
> +	// Capabilities is a list of capabilities that can be requested to add to the container.
> +	Capabilities []Capability `json:"capabilities,omitempty"`
> +	// HostPath determines if the policy allows containers to use the HostDir volume plugin
> +	HostPath bool `json:"hostPath,omitempty"`
> +	// HostNetwork determines if the policy allows the use of HostNetwork in the pod spec.
> +	HostNetwork bool `json:"hostNetwork,omitempty"`
> +	// HostPorts determines if the policy allows host ports in the containers.
> +	HostPorts bool `json:"hostPorts,omitempty"`
> +	// SELinuxContext is the strategy that will dictate what labels will be set in the SecurityContext.
> +	SELinuxContext SELinuxContextStrategyOptions `json:"seLinuxContext,omitempty"`

Are we framing this as an upper bound of allowed capabilities? a.k.a. Can I
create a pod with less capabilities than my PodSecurityPolicy would allow?
If so, the godoc should be changed.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/7893/files#r39452293>.